sources/packages/web/src/components/common/IdleTimeChecker.vue

```diff
@@ -57,44 +84,114 @@ export default {
     const resetIdleCheckerTimer = () => {
       clearInterval(interval.value);
       if (isAuthenticated.value) {
-        /* eslint-disable */
-        interval.value = setInterval(checkIdle, 30000);
-        /* eslint-enable */
+        interval.value = setInterval(checkIdle, 1000);
       }
     };
 
+    watch(
+      () => modalOpen.value,
+      (newValue: boolean) => {
+        if (newValue) {
+          countdownInterval.value = setInterval(() => {
+            if (countdown.value > 0) {
+              countdown.value = maximumIdleTime.value - idleTimeInSeconds.value;
+            }
+          }, 1000);
+        } else {
+          clearInterval(countdownInterval.value);
+          countdown.value = COUNT_DOWN_TIMER_FOR_LOGOUT;
+        }
+      },
+    );
+
+    const extendUserSessionTime = () => {
+      setLastActivityTime();
+      resetIdleCheckerTimer();
+      modalOpen.value = false;
+      clearInterval(countdownInterval.value);
+      countdown.value = COUNT_DOWN_TIMER_FOR_LOGOUT;
+    };
+
     const confirmExtendTimeModal = async () => {
+      modalOpen.value = true;
       if (await extendTimeModal.value.showModal()) {
-        lastActivityLogin.value = new Date();
-        resetIdleCheckerTimer();
+        extendUserSessionTime();
+      } else {
+        setLoggedOut();
       }
     };
 
     const checkIdle = () => {
-      const idleTimeInMintutes = getDatesDiff(
-        lastActivityLogin.value,
+      if (getLoggedOut()) {
+        logoff();
+      }
+      idleTimeInSeconds.value = getDatesDiff(
+        getLastActivityTime(),
         new Date(),
         "second",
-        true,
+        false,
       );
-      if (idleTimeInMintutes >= minimumIdleTime.value) {
+
+      // Exceeded user session time.
+      if (idleTimeInSeconds.value > maximumIdleTime.value) {
+        if (modalOpen.value === true) {
+          modalOpen.value = false;
+          clearInterval(countdownInterval.value);
```

Comment:
do we need this line, bcz line 138 will trigger the watch right? and watch alreday has these logic.

Reply:
Do you mean line 139, right?

Reply:
Anyways, I changed that part a bit.